### PR TITLE
HMX: add disable wake up source for in_start and digital in

### PR DESCRIPTION
**Pull request by RikardO-HM**

### CODE REVIEW

File: docs/interfaces/hmx/digital_io.md [hunk 1]
```diff
@@ -73,7 +73,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
 ## Sleep and wake up
 
 For Digital in, wake-up is default on. If you want to disable the inputs one way is to change the device tree when building the image, the other is to unbind the gpio-keys.
+Disable all input wake ups.
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:76]:
```suggestion
For Digital in, wake-up is enabled by default. If you want to disable this feature, one way is to modify the device tree when building the image, the other is to unbind the gpio-keys as follows.
```

File: docs/interfaces/hmx/digital_io.md [hunk 2]
```diff
@@ -73,7 +73,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
...
 ```
 echo gpio-keys > /sys/bus/platform/drivers/gpio-keys/unbind
 ```
-After waking up, you can bind it again if you want to use the gpio-keys.
\ No newline at end of file
+After waking up, you can bind it again if you want to use the gpio-keys.
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:80]:
```suggestion
After waking up, you can rebind if you want to use the gpio-keys.
```

File: docs/interfaces/hmx/digital_io.md [hunk 3]
```diff
@@ -73,7 +73,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
...
+
+Disable one or more wake up sources using using the macro code number for that signal
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:82]:
```suggestion
You can disable one or more wake-up sources using signal-specific macro codes.
```

File: docs/interfaces/hmx/digital_io.md [hunk 4]
```diff
@@ -73,7 +73,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
...
+
+| Hardware Name           | Software Name | Key Code | Macro Code |
+|-------------------------|---------------|----------|------------|
+| Digital_input_start     | in_start      | 656      | KEY_MACRO1	|
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:86]:
```suggestion
| Digital_input_start     | in_start      | 656      | KEY_MACRO1 |
```

File: docs/interfaces/hmx/digital_io.md [hunk 5]
```diff
@@ -73,7 +73,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
...
+| Digital_input_pulldown_1| in_pulldown1  | 657      | KEY_MACRO2 |
+| Digital_input_pulldown_2| in_pulldown2  | 658      | KEY_MACRO3 |
+| Digital_input_pullup_1  | in_pullup1    | 659      | KEY_MACRO4 |
+| Digital_input_pullup_2  | in_pullup2    | 660      | KEY_MACRO5 |
+| Digital_input_HMI_1     | in_hmi1       | 661      | KEY_MACRO6 |
+| Digital_input_HMI_2     | in_hmi2       | 662      | KEY_MACRO7 |
+
+Example disable in_start (all start signals)
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:94]:
```suggestion
Example on how to disable in_start (all start signals):
```

File: docs/interfaces/hmx/digital_io.md [hunk 6]
```diff
@@ -73,7 +73,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
...
+```
+echo 656 | tee /sys/bus/platform/drivers/gpio-keys/gpio-keys/disabled_keys
+```
+It is also possible within limit to see what cause the wake up.
+All wakeup_count (gpio-keys is wakeup4)
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:99]:
```suggestion
It is also possible, within limits, to see what caused the wake up.
The following displays all wakeup_count instances (gpio-keys is wakeup4)
```

Comment (albsod) [reply docs/interfaces/hmx/digital_io.md:99]: I'm not entirely sure what is meant by limit here.

Comment (RikardO-HM) [reply docs/interfaces/hmx/digital_io.md:99]:
```suggestion
It is also possible, within limits, to see what caused the wake up.
The following example displays all wakeup_count instances where gpio-keys is wakeup4 (wakeup4 has all digital in together so you can not see which signal that has been triggerd).
```

File: docs/interfaces/hmx/digital_io.md [hunk 7]
```diff
@@ -72,8 +72,30 @@ The unit has fault detection that goes high if a fault is detected(gpiod).
...
+| Digital_input_pullup_2  | in_pullup2    | 660      | KEY_MACRO5 |
+| Digital_input_HMI_1     | in_hmi1       | 661      | KEY_MACRO6 |
+| Digital_input_HMI_2     | in_hmi2       | 662      | KEY_MACRO7 |
+
+Example on how to disable in_start (all start signals):
+```
+echo 656 | tee /sys/bus/platform/drivers/gpio-keys/gpio-keys/disabled_keys
+```
+It is also possible, within limits, to see what caused the wake up.
+The following example displays all wakeup_count instances where gpio-keys is wakeup4 (wakeup4 has all digital in together so you can not see which signal that has been triggerd).
```
Comment (albsod) [inline docs/interfaces/hmx/digital_io.md:98]:
```suggestion
The following example displays all wakeup_count instances where gpio-keys is wakeup4 (wakeup4 has all digital in together so here you cannot see which signal was triggered).
```

**Review by albsod — APPROVED** [review]

**Pull request merged**